platform/BufferedSerial.cpp

```diff
@@ -94,6 +94,18 @@ int BufferedSerial::sync()
     return 0;
 }
 
+void BufferedSerial::sigio(Callback<void()> func) {
```

Comment:
The callback is normally called from interrupt.

Here we call it from the foreground registration point if the device is ready, and while that happens we can't afford for an interrupt that might call it to be triggered.